packages-ts/starknet-gauntlet-starkgate/test/commands/bridge.test.ts

```diff
@@ -0,0 +1,164 @@
+import { makeProvider } from '@chainlink/starknet-gauntlet'
+import deployOZCommand from '../../../starknet-gauntlet-oz/src/commands/account/deploy'
+import deployTokenCommand from '../../src/commands/token/deploy'
+import deployCommand from '../../src/commands/bridge/deploy'
+import setL1Bridge from '../../src/commands/bridge/setL1Bridge'
+import setL2Token from '../../src/commands/bridge/setL2Token'
+import {
+  registerExecuteCommand,
+  TIMEOUT,
+  LOCAL_URL,
+  startNetwork,
+  IntegratedDevnet,
+} from '@chainlink/starknet-gauntlet/test/utils'
+import { loadContract, CONTRACT_LIST } from '../../src/lib/contracts'
+import { Contract } from 'starknet'
+import { BN } from '@chainlink/gauntlet-core/dist/utils'
+import { compressProgram } from 'starknet/dist/utils/stark'
+import fetch from 'node-fetch'
+
+describe('Bridge Contract', () => {
+  let network: IntegratedDevnet
+  let account: string
+  let privateKey: string
+  let publicKey: string
+  let l1BridgeAddress: number = 42 // mock placeholder
+  let bridgeContractAddress: string
+  let tokenContractAddress: string
+
+  beforeAll(async () => {
+    network = await startNetwork()
+  }, 5000)
+
+  it(
+    'Deploy OZ Account',
+    async () => {
+      const command = await registerExecuteCommand(deployOZCommand).create({}, [])
+
+      const report = await command.execute()
+      expect(report.responses[0].tx.status).toEqual('ACCEPTED')
+
+      account = report.responses[0].contract
+      privateKey = report.data.privateKey
+      publicKey = report.data.publicKey
+
+      // Fund the newly allocated account
+      let gateway_url = process.env.NODE_URL || 'http://localhost:5050'
+      let balance = 100_000_000_000_000
+      // let balance = 1e21
+      const body = {
+        address: account,
+        amount: balance,
+        lite: true,
+      }
+      const response = await fetch(`${gateway_url}/mint`, {
+        method: 'post',
+        body: JSON.stringify(body),
+        headers: { 'Content-Type': 'application/json' },
+      })
```

Comment:
Does starknet.js have any support for this? If no, could we extract this to `starknet-gauntlet`? I could see this function being used in almost every test

Reply:
Yeah it's specific to devnet: https://github.com/Shard-Labs/starknet-devnet#mint-token---local-faucet=

There's `fetchEndpoint` but it's `protected` so we can't use it :/ https://github.com/0xs34n/starknet.js/blob/8969decf73cb62096da8804c37be135f624ac777/src/provider/default.ts#L140-L197=

Rather than deploying our own account, we could use a specific seed for the devnet, then use one of those 10 accounts that are pre-allocated.

Reply:
Definitely, that'd be perfect

Reply:
tbh I prefer deploying our own account just to make sure that command works - also if the starknet-devnet development lags behind contract develop...not sure if any issues could show up there